core/startos/src/service/service_effect_handler.rs

```diff
@@ -343,17 +403,52 @@ async fn export_service_interface(
 async fn get_primary_url(
     context: EffectContext,
     data: GetPrimaryUrlParams,
-) -> Result<Value, Error> {
-    todo!()
+) -> Result<HostAddress, Error> {
```

Comment:
I feel like this should return `Option<HostAddress>`. Does that not match the sdk?

Reply:
The base branch (next/minor) has [getPrimaryUrl](https://github.com/Start9Labs/start-os/blob/next/minor/sdk/lib/types.ts#L422-L426
) returning a `Promise<UrlString | null>`, where `UrlString` is an alias for `string` - This is consistent with the `address` field of [HostAddress](https://github.com/Start9Labs/start-os/blob/next/minor/core/startos/src/net/host/address.rs#L5-L14).

So I believe the return type should actually be updated to `<Option<String>`, correct? Or would this need to be wrapped in a `Result` to handle errors?